test/test_datatype.jl

```diff
@@ -158,6 +158,10 @@ end
     RowType = MPI.Types.create_vector(8, 1, 8, MPI.DOUBLE)
     @test typeof(RowType) == MPI.Datatype
 
+    # create_hvector
+    DType = MPI.Types.create_hvector(8, 1, 8, MPI.DOUBLE)
+    @test typeof(DType) == MPI.Datatype
```

Comment:
Maybe add a test that calls `Types.size` and checks it is the correct size?:
https://github.com/JuliaParallel/MPI.jl/blob/73139c57ac9f8d956c8c45e9a632a4b008114f40/src/datatypes.jl#L170